base/parse.jl

```diff
@@ -102,8 +102,9 @@ end
         _a <= _c <= _z ? _c-_a+a           : UInt32(base)
 end
 
-
-function tryparse_internal(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, base_::Integer, raise::Bool) where T<:Integer
+# `@constprop :aggressive` here to make sure we const-prop' `raise` argument to obtain
+# the return-type stability for the `raise === true` case
+@constprop :aggressive function tryparse_internal(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, base_::Integer, raise::Bool) where T<:Integer
```

Comment:
If `raise` is that important for type stability, wouldn't it be better to lift it into the type domain? `@constprop :aggressive` on this function seems quite arbitrary.

Reply:
Sounds reasonable. Lifted it to the type domain.